portals/publisher/source/src/app/components/Apis/Details/Configuration/components/AdvertiseInfo.jsx

```diff
@@ -0,0 +1,353 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { FormattedMessage } from 'react-intl';
+import { isRestricted } from 'AppData/AuthManager';
+import { useAPI } from 'AppComponents/Apis/Details/components/ApiContext';
+import MuiAlert from 'AppComponents/Shared/MuiAlert';
+import { useRevisionContext } from 'AppComponents/Shared/RevisionContext';
+import TextField from '@material-ui/core/TextField';
+import Box from '@material-ui/core/Box';
+import Grid from '@material-ui/core/Grid';
+import FormControlLabel from '@material-ui/core/FormControlLabel';
+import { makeStyles } from '@material-ui/core/styles';
+import FormLabel from '@material-ui/core/FormLabel';
+import RadioGroup from '@material-ui/core/RadioGroup';
+import Radio from '@material-ui/core/Radio';
+import FormControl from '@material-ui/core/FormControl';
+import HelpOutline from '@material-ui/icons/HelpOutline';
+import Tooltip from '@material-ui/core/Tooltip';
+
+const useStyles = makeStyles((theme) => ({
+    expansionPanel: {
+        marginBottom: theme.spacing(3),
+    },
+    expansionPanelDetails: {
+        flexDirection: 'column',
+    },
+    iconSpace: {
+        marginLeft: theme.spacing(0.5),
+    },
+    actionSpace: {
+        margin: '-7px auto',
+    },
+    subHeading: {
+        margin: 0,
+        display: 'inline-flex',
+        lineHeight: 1.5,
+        paddingBottom: '20px',
+    },
+    mandatoryStar: {
+        color: theme.palette.error.main,
+        marginLeft: theme.spacing(0.1),
+    },
+    alert: {
+        marginBottom: '20px',
+    },
+}));
+
+/**
+ *
+ *
+ * @export
+ * @param {*} props
+ * @returns
+ */
+const AdvertiseInfo = (props) => {
+    const {
+        configDispatcher,
+        oldApi: { policies: oldPolicies, endpointConfig: oldEndpointConfig },
+        api: {
+            advertiseInfo,
+            type,
+            policies,
+            lifeCycleStatus,
+            endpointConfig,
+        },
+        setIsOpen,
+    } = props;
+    const {
+        allRevisions,
+    } = useRevisionContext();
+    const classes = useStyles();
+    const [apiFromContext] = useAPI();
+    const isDeployed = () => {
+        if (allRevisions) {
+            for (let i = 0; i < allRevisions.length; i++) {
+                if (allRevisions[i].deploymentInfo.length > 0) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    };
+    const infoMsg = () => {
+        if (type === 'ASYNC') {
```

Comment:
remove the method to and add it to the component

Reply:
done with https://github.com/wso2/apim-apps/pull/130/commits/8958439e3142cac7c09571f13a7fbf782e21aba2